### ISSUE TITLE
Klasifikacija umetničkih dela po umetničkom pravcu

### ISSUE THREAD

**Issue by sofijam2011** [ftn-ai-lab/sc-2025#23]
1. Tim
Dragana Kanazir RA 11/2022 i Sofija Mihajlović RA 10/2022.

2. Asistent
Teodor Vidaković.

3. Definicija problema
Automatska klasifikacija umetničkih dela u odgovarajući umetnički pravac - renesansa, barok, impresionizam, kubizam, itd. na osnovu vizuelnih karakteristika slike korišćenjem neuronskih mreža.

4. Motivacija problema
Razvoj ovog sistema ima višestruku praktičnu primenu:
- Edukativne aplikacije - pomoć posetiocima muzeja da automatski prepoznaju stil umetničkog dela skeniranjem telefonom.
- Digitalne umetničke kolekcije - automatsko tagovanje i kategorizacija hiljada umetničkih dela.
5. Skup podataka
Pronašle smo na ovoj stranici: https://universe.roboflow.com/art-dataset/wiki-art/browse?queryText=&pageSize=50&startingIndex=0&browseQuery=true 
Izbor pravaca za projekat (7 umetničkih pravaca):
1.Renesansa (Renaissance) - klasični realizam, perspektiva, religiozne teme
2.Barok (Baroque) - drama, kontrast svetla i tame 
3.Impresionizam (Impressionism) - fokus na svetlost
4.Post-Impresionizam (Post-Impressionism) - ekspresivnije boje, Van Gogh stil
5.Kubizam (Cubism) - geometrijske forme, multiple perspektive
6.Apstraktni ekspresionizam (Abstract Expressionism) - potpuno apstraktno, color fields
7.Pop Art - popularna kultura, jarke boje, komercijalni stil
Oni koji ne pripadaju nijednom od ovih pravaca bili pripadali bi nesvrstanim.

6. Metodologija
Sve slike umetničkih dela ćemo prvo skalirati na istu veličinu (224×224 piksela) i normalizovati vrednosti piksela na opseg [0,1]. Primenićemo dva pristupa dubokog učenja bazirana na konvolucionim neuronskim mrežama:
Pristup 1 - CNN sa nule:
Dizajniraćemo konvolucionu neuronsku mrežu koja se sastoji od tri konvoluciona bloka. Prvi blok sadrži Conv2D sloj sa 32 filtera (3×3 kernel) i ReLU aktivacijom, nakon čega sledi MaxPooling2D (2×2). Drugi blok ima Conv2D sloj sa 64 filtera i MaxPooling, dok treći blok koristi 128 filtera sa istom strukturom. Nakon konvolucionih blokova primenjujemo Flatten sloj, zatim dva Dense sloja (256 i 128 neurona) sa ReLU aktivacijom i Dropout regularizacijom (0.3) između njih. Izlazni sloj ima 7 neurona sa softmax aktivacijom za klasifikaciju u jedan od sedam umetničkih pravaca. Model treniramo sa Adam optimizerom i categorical crossentropy loss funkcijom.
Pristup 2 - Transfer Learning sa MobileNetV2:
Koristićemo MobileNetV2 model pretreniran na ImageNet datasetu. Bazne slojeve pretreniranog modela zamrzavamo, a dodajemo prilagođenu klasifikacijsku glavu koja se sastoji od GlobalAveragePooling2D sloja, Dense sloja sa 256 neurona i ReLU aktivacijom, Dropout sloja (0.5) i finalnog Dense sloja sa 7 neurona i softmax aktivacijom. Model treniramo sa Adam optimizerom i nižim learning rate-om (0.0001) kako bi se postojeće naučene feature-e prilagodile specifičnostima umetničkih dela.
Oba pristupa koristiće data augmentation tehnike (horizontalno okretanje, rotacija, zoom) kako bi se poboljšala generalizacija modela.

7. Evaluacija
Podela dataseta na 70% trening, 15% validacija i 15% test.
Metrike evaluacije:
Accuracy: Procenat tačno klasifikovanih slika (cilj: preko 80%)
F1 score: Balansirana metrika koja uzima u obzir precision i recall, posebno važna ako dataset nije savršeno balansiran
Confusion Matrix: Vizuelni prikaz koji pokazuje koje umetničke pravce model najčešće meša između sebe (npr. da li model često konfuzuje Renesansu sa Barokom ili Impresionizam sa Post-impresionizmom), što može ukazati na vizuelne sličnosti između stilova
Precision i Recall po klasama: Za svaki od 7 umetničkih pravaca posebno, kako bismo videli koji stilovi se najbolje klasifikuju

**Comment by Tekisha** [comment]
- U sekciji metodologija, predloženi pristup nije dovoljno kompleksan. Istražite moguće pristupe dubokog učenja koji bi se mogli primeniti na ovaj problem (npr. CNN, transfer learning..) i izmenite predlog. S obzirom da radite u timu, potrebno je da u metodologiji navedete ***dva pristupa dubokog učenja***.

- Što se tiče evaluacije, istražite da li bi ***accuracy*** i ***confusion matrix*** dodatno pomogli u evaluaciji samog modela.

Naznačite komentarom kada promenite tekst specifikacije projekta.

**Comment by sofijam2011** [comment]
Poštovani, promenjen je tekst specifikacije projekta.

**Comment by Tekisha** [comment]
Tema odobrena.
Ostavite link do GitHub repozitorijuma u komentaru.
Srećan rad Vam želim!

**Comment by sofijam2011** [comment]
Link do repozitorijuma: https://github.com/draganak94/Klasifikacija_umetnickih_dela_po_umetnickom_pravcu
